tests/prepare_test_data.py

```diff
@@ -0,0 +1,173 @@
+        'skip-dirs': ['Godeps'],
+    },
+    {
+        'name': 'github.com/influxdata/influxdb',
+        'clone-url': 'https://github.com/influxdata/influxdb',
+        'commits': [
+            'f326c6c0e4349a6f9bca056571b01ded81a2887c',
+            '688bc7a2f1877b524a3c915ae1ff7981267fcb49',
+            'd771eb72f708c523bb2bec35a97f6bb83ae4ef1d',
+        ],
+        'skip-dirs': [],
+    },
+    {
+        'name': 'github.com/google/martian',
+        'clone-url': 'https://github.com/google/martian',
+        'commits': [
+            '4d4cb10a32e7f1237228d58ecf95c571b0947a11',
+            '1afb734cc2eb3a5dab708a901ee07106a94a9186',
+        ],
+        'skip-dirs': [],
+    },
+]
+
+
+def get_logger():
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    formatter = logging.Formatter('%(levelname)s: %(message)s')
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    return logger
+
+
+def prepare(destdir, projects):
+    logger = get_logger()
+    for project in projects:
+        logger.info('Preparing project "{}"'.format(project['name']))
+        basepath = os.path.join(destdir, project['name'])
+        try:
+            os.makedirs(basepath)
+        except OSError:
+            logger.error('Cannot create directory "{}"'.format(basepath))
+            continue
+        for commit in project['commits']:
+            logger.info('Processing commit "{}"'.format(commit))
+            path = os.path.join(basepath, commit)
+            data = {
+                'source_code_directory': path,
+                'directories_to_skip': project['skip-dirs'],
+                'project': project['name'],
+                'commit': commit,
+                'ipprefix': project['name'],
+            }
+            datafile = '{}.json'.format(path)
+            try:
+                with open(datafile, 'wb') as f:
+                    json.dump(data, f)
+            except IOError:
+                logger.error('Cannot open file "{}"'.format(datafile))
+                continue
+            try:
+                repo = git.Repo.clone_from(project['clone-url'], path)
+            except git.exc.GitCommandError:
+                logger.error('Cannot clone git repository "{}" to "{}"'.format(
+                    project['clone-url'], path))
+                continue
+            try:
+                archive = '{}.tar.gz'.format(path)
+                prefix = os.path.join(commit, '')
+                try:
+                    with open(archive, 'wb') as f:
+                        repo.archive(f, commit, prefix, format='tar.gz')
+                except (IOError, git.exc.GitCommandError):
+                    logger.error('Cannot create archive "{}"'.format(archive))
+                    continue
+                try:
+                    plugin = GoSymbolExtractor()
+                    if not plugin.setData(data):
```

Comment:
At this point, data's source_code_directory is set to path. Is the git repository in path checked out to data's commit?